src/main/java/org/broadinstitute/hellbender/tools/spark/utils/IntHistogram.java

```diff
@@ -0,0 +1,233 @@
+
+        if ( nBins * nUntrackedObservations >= totalObservations ) {
+            return this;
+        }
+
+        do {
+            nUntrackedObservations += counts[nBins - 2];
+            nBins -= 1;
+        } while ( nBins * nUntrackedObservations < totalObservations );
+
+        nBins += 1;
+        nUntrackedObservations -= counts[nBins - 2];
+        final long[] newCounts = Arrays.copyOfRange(counts, 0, nBins);
+        newCounts[nBins - 1] = nUntrackedObservations;
+        return new IntHistogram(newCounts, totalObservations);
+    }
+
+    public CDF getCDF() { return new CDF(this); }
+
+    public final static class Serializer extends com.esotericsoftware.kryo.Serializer<IntHistogram> {
+        @Override
+        public void write( final Kryo kryo, final Output output, final IntHistogram histogram ) {
+            histogram.serialize(kryo, output);
+        }
+
+        @Override
+        public IntHistogram read( final Kryo kryo, final Input input, final Class<IntHistogram> klass ) {
+            return new IntHistogram(kryo, input);
+        }
+    }
+
+    @DefaultSerializer(CDF.Serializer.class)
+    public final static class CDF {
+        final float[] cdfFractions;
+        final long nCounts;
+
+        public CDF( final IntHistogram intHistogram ) {
+            final int arrayLen = intHistogram.getMaximumTrackedValue() + 2;
+            cdfFractions = new float[arrayLen];
+            this.nCounts = intHistogram.getTotalObservations();
+            long sum = 0;
+            for ( int observedValue = 0; observedValue != arrayLen; ++observedValue ) {
+                sum += intHistogram.getNObservations(observedValue);
+                cdfFractions[observedValue] = (float)sum / nCounts;
+            }
+        }
+
+        private CDF( final Kryo kryo, final Input input ) {
+            final int len = input.readInt();
+            cdfFractions = new float[len];
+            for ( int idx = 0; idx != len; ++idx ) {
+                cdfFractions[idx] = input.readFloat();
+            }
+            nCounts = input.readLong();
+        }
+
+        private void serialize( final Kryo kryo, final Output output ) {
+            output.writeInt(cdfFractions.length);
+            for ( final float val : cdfFractions ) {
+                output.writeFloat(val);
+            }
+            output.writeLong(nCounts);
+        }
+
+        public IntHistogram createEmptyHistogram() {
+            return new IntHistogram(cdfFractions.length - 2);
+        }
+
+        /** Is the specified histogram significantly different (at the p < .01 level) from the CDF. */
+        public boolean isDifferent( final IntHistogram intHistogram ) { return isDifferent(intHistogram, .01f); }
+
+        /** Using the Kolmogorov-Smirnov statistic for two samples:
+         *  Is the specified histogram significantly different from the CDF. */
+        public boolean isDifferent( final IntHistogram sampleHistogram, final float significance ) {
```

Comment:
Do you have a sense of what size indels are causing the test to return true under your chosen parameters? You mentioned before that you thought it might be ringing the bell even for very small indels.